templates/admin/settings.tpl

```diff
@@ -12,54 +12,72 @@
 {assign var="uuid" value=""|uniqid|escape}
 <div id="settings-admin-{$uuid}">
 	<tabs>
+		{if $componentAvailability['SiteSetup']}
 		<tab id="setup" label="{translate key="admin.siteSetup"}">
 			<tabs :is-side-tabs="true">
+				{if $componentAvailability['SiteConfig']}
 				<tab id="settings" label="{translate key="admin.settings"}">
 					<pkp-form
 						v-bind="components.{$smarty.const.FORM_SITE_CONFIG}"
 						@set="set"
 					/>
 				</tab>
+				{/if}
+				{if $componentAvailability['SiteInfo']}
 				<tab id="info" label="{translate key="manager.setup.information"}">
 					<pkp-form
 						v-bind="components.{$smarty.const.FORM_SITE_INFO}"
 						@set="set"
 					/>
 				</tab>
+				{/if}
+				{if $componentAvailability['Languages']}
 				<tab id="languages" label="{translate key="common.languages"}">
 					{capture assign=languagesUrl}{url router=$smarty.const.ROUTE_COMPONENT component="grid.admin.languages.AdminLanguageGridHandler" op="fetchGrid" escape=false}{/capture}
 					{load_url_in_div id="languageGridContainer" url=$languagesUrl}
 				</tab>
+				{/if}
+				{if $componentAvailability['NavigationMenus']}
 				<tab id="nav" label="{translate key="manager.navigationMenus"}">
 					{capture assign=navigationMenusGridUrl}{url router=$smarty.const.ROUTE_COMPONENT component="grid.navigationMenus.NavigationMenusGridHandler" op="fetchGrid" escape=false}{/capture}
 					{load_url_in_div id="navigationMenuGridContainer" url=$navigationMenusGridUrl}
 					{capture assign=navigationMenuItemsGridUrl}{url router=$smarty.const.ROUTE_COMPONENT component="grid.navigationMenus.NavigationMenuItemsGridHandler" op="fetchGrid" escape=false}{/capture}
 					{load_url_in_div id="navigationMenuItemsGridContainer" url=$navigationMenuItemsGridUrl}
 				</tab>
+				{/if}
 				{call_hook name="Template::Settings::admin::setup"}
 			</tabs>
 		</tab>
+		{/if}
+		{if $componentAvailability['SiteAppearance']}
 		<tab id="appearance" label="{translate key="manager.website.appearance"}">
 			<tabs :is-side-tabs="true">
+				{if $componentAvailability['SiteTheme']}
 				<tab id="theme" label="{translate key="manager.setup.theme"}">
 					<theme-form
 						v-bind="components.{$smarty.const.FORM_THEME}"
 						@set="set"
 					/>
 				</tab>
+				{/if}
+				{if $componentAvailability['SiteAppearanceSetup']}
 				<tab id="setup" label="{translate key="navigation.setup"}">
 					<pkp-form
 						v-bind="components.{$smarty.const.FORM_SITE_APPEARANCE}"
 						@set="set"
 					/>
 				</tab>
+				{/if}
 				{call_hook name="Template::Settings::admin::appearance"}
 			</tabs>
 		</tab>
+		{/if}
+		{if $componentAvailability['SitePlugins']}
 		<tab id="plugins" label="{translate key="common.plugins"}">
 			{capture assign=pluginGridUrl}{url router=$smarty.const.ROUTE_COMPONENT component="grid.admin.plugins.AdminPluginGridHandler" op="fetchGrid" escape=false}{/capture}
 			{load_url_in_div id="pluginGridContainer" url=$pluginGridUrl}
 		</tab>
+		{/if}
```

Comment:
I wonder if we need this level of abstraction in the template. Wouldn't it be enough to have a single template boolean, `hasMultipleContexts`, and use that to hide the tabs we don't want?

With the component availability approach, it's possible to accidentally get yourself into a case where a parent tab is meant to be hidden but a child tab is meant to be shown.

But I'm ok leaving it as it is if you think it's better this way.

Reply:
I regard the availability of the tabs to be a "business rule" and I tend to add the business rules in the handler (keeping the template as clean as possible). 

Also I thought that we could somehow pass this array to the corresponding tabs, thus erasing the multiple `{if}` statements.

> With the component availability approach, it's possible to accidentally get yourself into a case where a parent tab is meant to be hidden but a child tab is meant to be shown.
My initial approach was to actually create a structure holding the data connections that describe the parent-child relationships, and add the code that would fix that. But I thought I would add unnecessary complexity, given that the template it self would not use that structure. Also someone could argue about the hierarchy of things (which availability has the priority, parent's or child's? Effectively the parent's availability priority has been chosen. One drawback of that is that we could end-up with an empty tab - Maybe that's something that needs fixing.

Reply:
Ok, let's leave it as-is.